generator/pkg/schemagen/generate.go

```diff
@@ -588,7 +660,17 @@ func (g *schemaGenerator) isPartOfOwnPackage(t reflect.Type) bool {
 }
 
 func (g *schemaGenerator) resolveJavaClassUsingOwnPackages(t reflect.Type) string {
-	return g.packageMapping[t.PkgPath()].JavaPackage + "." + t.Name()
+	return g.adaptJavaClassName(g.packageMapping[t.PkgPath()].JavaPackage + "." + t.Name())
+}
+
+func (g *schemaGenerator) adaptJavaClassName(className string) string {
```

Comment:
@metacosm @manusa I've created a new method where all the java classes (including the interfaces) will use to adapt the final java name.
The only rule in place will modify names like `is_Something` to `Is_Something`. 
I would happy to also add a new rule to also change from `Is_Something` to `IsSomething` but I'm afraid that this can bring breaking changes to existing extensions. Wdyt?

Reply:
We shouldn't break existing extensions at this point but maybe that's something to consider for 6.0?

Reply:
If you're talking about a jsonschema rule here, I think it's better not too.

We should try to avoid changes to the current Extensions APIs.

Also we need to keep in mind, in the (near) future, both generators should be merged, so every change added to this generators should be considered when merging both.

Reply:
@manusa at the moment, the only rule is to parse java names from is_Class to Is_Class. The rest is to allow extensions adapt their model with custom java names rule (something we need for the Istio extension). So, I think it's ok and it won't break anything. Anyway, let me know if you want to change something from this PR before being merged.